web/packages/teleport/src/SessionRecordings/view/player/PlayerControls.tsx

```diff
@@ -0,0 +1,376 @@
+      bg="levels.surface"
+      borderTop="1px solid"
+      borderColor="spotBackground.0"
+      height={CONTROLS_HEIGHT}
+      px={2}
+      flexShrink={0}
+      ref={containerRef}
+    >
+      {onToggleSidebar && (
+        <HoverTooltip
+          tipContent="Toggle Sidebar"
+          portalRoot={containerRef.current}
+        >
+          <PlayerButton onClick={onToggleSidebar}>
+            <Layout size="small" />
+          </PlayerButton>
+        </HoverTooltip>
+      )}
+
+      <PlayButton
+        disabled={buttonDisabled}
+        onClick={handlePlayerButtonClick}
+        state={state}
+        containerRef={containerRef}
+      />
+
+      <Box
+        fontFamily="mono"
+        fontSize="small"
+        pl={2}
+        pr={4}
+        ref={durationRef}
+        width="70px"
+      >
+        0:00
+      </Box>
+
+      <ProgressBarArea
+        onClick={handleProgressBarClick}
+        onMouseLeave={handleProgressBarMouseLeave}
+        onMouseMove={handleProgressBarMouseMove}
+      >
+        <ProgressBarContainer ref={progressBarContainerRef}>
+          <SeekingBar ref={seekingRef} />
+          <ProgressBar ref={progressBarRef} />
+        </ProgressBarContainer>
+      </ProgressBarArea>
+
+      {onToggleTimeline && (
+        <HoverTooltip
+          tipContent="Toggle Timeline"
+          portalRoot={containerRef.current}
+        >
+          <PlayerButton onClick={onToggleTimeline}>
+            <FilmStrip size="small" />
+          </PlayerButton>
+        </HoverTooltip>
+      )}
+
+      {onToggleFullscreen && (
+        <HoverTooltip
+          tipContent={fullscreen ? 'Exit full screen' : 'Full Screen'}
+          portalRoot={containerRef.current}
+        >
+          <PlayerButton onClick={onToggleFullscreen}>
+            {fullscreen ? (
+              <CornersIn size="small" />
+            ) : (
+              <CornersOut size="small" />
+            )}
+          </PlayerButton>
+        </HoverTooltip>
+      )}
+    </Flex>
+  );
+}
+
+interface PlayButtonProps {
+  disabled: boolean;
+  onClick: () => void;
+  state: PlayerState;
+  containerRef: RefObject<HTMLDivElement>;
+}
+
+function PlayButton({
+  disabled,
+  onClick,
+  state,
+  containerRef,
+}: PlayButtonProps) {
+  const { icon: Icon, text } = useMemo(() => {
+    switch (state) {
+      case PlayerState.Playing:
+        return { icon: CirclePause, text: 'Pause' };
+      case PlayerState.Paused:
+        return { icon: CirclePlay, text: 'Play' };
+      case PlayerState.Stopped:
+        return { icon: Refresh, text: 'Replay' };
+      case PlayerState.Loading:
+        return { icon: SpinningSpinner };
+    }
+  }, [state]);
+
+  const button = (
+    <PlayerButton disabled={disabled} onClick={onClick}>
+      <Icon size="small" />
+    </PlayerButton>
+  );
+
+  if (text) {
+    return (
+      <HoverTooltip tipContent={text} portalRoot={containerRef.current}>
+        {button}
+      </HoverTooltip>
+    );
+  }
```

Comment:
`HoverTooltip` supports the case where `tipContent` is empty or undefined, so we don't have to make this a conditional.

Reply:
nice, cheers